### PR TITLE
Implement sha256 and keccak hooks

### PR DESCRIPTION
**Pull request by yzhang90**
* add k-blockchain-plugin. The submodule contains krypto.md, which implements crypto functions like Keccak and Sha256.
* implement sha256 elrond hook
* implement keccak elrond hook
* add two basic feature tests to test above two hooks.

### CODE REVIEW

File: elrond.md [hunk 1]
```diff
@@ -936,6 +969,40 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
          </locals>
 ```
 
+### Crypto EI
```
Comment (hjorthjort) [inline elrond.md:972]: What does EI mean?

Comment (yzhang90) [reply elrond.md:972]: no idea. it just appear in the their filename. I'll remove it

File: elrond.md [hunk 2]
```diff
@@ -918,6 +951,40 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
...
+               ~> #memStoreFromBytesStack(RESULTOFFSET)
+               ~> #dropBytes
+               ~> i32.const 0
+                  ...
+         </instrs>
+         <locals>
+           0 |-> <i32> DATAOFFSET
+           1 |-> <i32> LENGTH
+           2 |-> <i32> RESULTOFFSET
+         </locals>
```
Comment (hjorthjort) [inline elrond.md:985]: Looks like you could easily simplify these two rules into one:

```
syntax InternalInstrs ::= #hashMemory ( HashBytesStackInstr )
```
where the `HashBytesStackInstr` is either `#sha256FromBytesStack` or `#keccakFromBytesStack`

**Review by hjorthjort — CHANGES_REQUESTED** [review]
Small change and a question.

Also, can you give a short explanation of the PR in the description? Even if I know what's going on, it's good for people looking back at the history to get a sense of what's going on.

File: Makefile [hunk 3]
```diff
@@ -57,35 +94,50 @@ elrond-contracts:
...
-                         wasm-coverage
-EXTRA_FILES           := $(patsubst %,%.md,$(EXTRA_FILE_NAMES))
-EXTRA_FILES_KWASM_DIR := $(patsubst %,$(KWASM_SUBMODULE)/%.md,$(EXTRA_FILE_NAMES))
+ELROND_FILE_NAMES      := elrond        \
+                          mandos        \
+                          wasm-coverage
+PLUGIN_FILE_NAMES      := blockchain-k-plugin/krypto
+EXTRA_SOURCES          := $(patsubst %,%.md,$(ELROND_FILE_NAMES) $(PLUGIN_FILE_NAMES))
+ELROND_FILES_KWASM_DIR := $(patsubst %,$(KWASM_SUBMODULE)/%.md,$(ELROND_FILE_NAMES))
+PLUGIN_FILES_KWASM_DIR := $(patsubst %,$(KWASM_SUBMODULE)/%.md,$(PLUGIN_FILE_NAMES))
```
[the remark this answers is not in the document]
Comment (yzhang90) [reply Makefile:117]: see the description above. Basically, in order to implement the kccak and sha256 hooks in elrond, we need to use the functions defined in krypto.md. The file is now located inthe k-blockain-plugin submodule.

**Review by hjorthjort — APPROVED** [review]

**Pull request merged**